libyul/backends/wasm/EVMToEwasmTranslator.cpp

```diff
@@ -812,13 +812,16 @@ function keccak256(x1, x2, x3, x4, y1, y2, y3, y4) -> z1, z2, z3, z4 {
 }
 
 function address() -> z1, z2, z3, z4 {
-	eth.getAddress(0:i32)
-	z1, z2, z3, z4 := mload_internal(0:i32)
+	eth.getAddress(4:i32)
+	z2 := i64.and(bswap64(i64.load(0:i32)), 0x00000000ffffffff)
```

Comment:
Why not `i32.load(4:i32)`?

Reply:
Addresses are defined being the rightmost 160 bits of the keccak hash. So I just write the address with a 4 bytes offset. I don't touch `z1` (return values seem to be always zero initialized), but load `z2` from offset 0, where it will have already the needed bytes shifted by 32 bits.

Reply:
Alright!